src/vm/object_model.rs

```diff
@@ -488,17 +488,33 @@ pub mod specs {
     // This macro is invoked in define_vm_metadata_global_spec or define_vm_metadata_local_spec.
     // Use those two to define a new VM metadata spec.
     macro_rules! define_vm_metadata_spec {
-        ($spec_name: ident, $is_global: expr, $log_num_bits: expr, $side_min_obj_size: expr) => {
+        ($(#[$outer:meta])*$spec_name: ident, $is_global: expr, $log_num_bits: expr, $side_min_obj_size: expr) => {
+            $(#[$outer])*
             pub struct $spec_name(MetadataSpec);
             impl $spec_name {
+                #[doc="The number of bits (in log2) that are needed for the spec."]
                 pub const LOG_NUM_BITS: usize = $log_num_bits;
+
+                #[doc="Whether this spec is global or local. The binding needs to make sure"]
+                #[doc="global specs are laid out after a global spec, and local specs are laid"]
```

Comment:
I think the notion of "laid out after" only applies for side metadata.  In header metadata only takes a offset, therefore the order doesn't matter.  We should mention "side metadata" here.

Reply:
Done